Replace checks for ending slash in file paths with runtime assertions

LumioseSil opened the pull request:
#### cb1f64337310fec41266ad91ef5600b6a45808c5
<pre>
Replace checks for ending slash in file paths with runtime assertions
<a href="https://bugs.webkit.org/show_bug.cgi?id=253253">https://bugs.webkit.org/show_bug.cgi?id=253253</a>

Reviewed by Alexey Proskuryakov.

This issue has been long fixed on iOS Simulator. Therefore, this check
is safe to turn into a runtime assertion.

* Source/WTF/wtf/cocoa/FileSystemCocoa.mm:(openTemporaryFile): Change
  temporaryFilePath.last() check to a runtime assertion.

* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:(createHandleForTemporaryFile): Ditto.

Canonical link: <a href="https://commits.webkit.org/261246@main">https://commits.webkit.org/261246@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2a0a413ebe0a8aa282b0332f5f3013672e61610f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111039 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20179 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43606 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/2467 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119884 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21554 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11281 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/2110 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116793 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16008 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99192 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/103545 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97965 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30856 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44462 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/99630 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12696 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/32191 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/86370 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/10809 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13239 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9173 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/100763 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18656 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/51792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31416 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7799 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15188 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/108800 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26822 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WTF/wtf/cocoa/FileSystemCocoa.mm:
```diff
@@ -104,10 +104,8 @@ String openTemporaryFile(StringView prefix, PlatformFileHandle& platformFileHand
-    // fixed in all iOS Simulator versions that we use.
-    if (temporaryFilePath.last() != '/')
-        temporaryFilePath.append('/');
+    // Paths must end with a slash.
```
Please remove this comment. It doesn't add anything at all to the actual code.

aproskuryakov commented on Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
```diff
@@ -280,11 +280,9 @@ String resolvePathForSandboxExtension(StringView path)
-    if (path.last() != '/')
-        path.append('/');
-    
+    // Paths must end with a slash.
```
Ditto.

aproskuryakov reviewed (changes requested):
Looks like it's believed to be fixed indeed, thank you for catching that, and for the PR.

LumioseSil committed:
Replace checks for ending slash in file paths with runtime assertions
https://bugs.webkit.org/show_bug.cgi?id=253253

Reviewed by Alexey Proskuryakov.

This issue has been long fixed on iOS Simulator. Therefore, this check
is safe to turn into a runtime assertion.

* Source/WTF/wtf/cocoa/FileSystemCocoa.mm:(openTemporaryFile): Change
  temporaryFilePath.last() check to a runtime assertion.

* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:(createHandleForTemporaryFile): Ditto.

Canonical link: https://commits.webkit.org/261246@main
